src/jaxsim/parsers/kinematic_graph.py

```diff
@@ -452,58 +533,77 @@ def reduce(self, considered_joints: List[str]) -> KinematicGraph:
         # 4. Resolve the pose of the frames wrt their reduced graph parent
         # ================================================================
 
-        # Update frames properties using the transforms from the full graph
-        for frame in reduced_graph.frames:
-            # Get the link in which the removed link was lumped into
-            new_parent_link = links_dict[frame.name]
+        # Build a new object to compute FK on the reduced graph.
+        fk_reduced = KinematicGraphTransforms(graph=reduced_graph)
 
-            msg = f"New parent of frame '{frame.name}' is '{new_parent_link.name}'"
-            logging.info(msg)
+        # We need to adjust the pose of the frames since their parent link
+        # could have been removed by the reduction process.
+        for frame in reduced_graph.frames:
 
-            # Update the connection of the frame
-            frame.parent = new_parent_link
-            frame.pose = fk.relative_transform(
-                relative_to=new_parent_link.name, name=frame.name
+            # Always find the real parent link of the frame
+            name_of_new_parent_link = fk_reduced.find_parent_link_of_frame(
+                name=frame.name
             )
+            assert name_of_new_parent_link in reduced_graph, name_of_new_parent_link
+
+            # Notify the user if the parent link has changed.
+            if name_of_new_parent_link != frame.parent.name:
+                msg = "New parent of frame '{}' is '{}'"
+                logging.debug(msg.format(frame.name, name_of_new_parent_link))
```

Comment:
Minor issue, but it's to be coherent with the library style
```suggestion
                logging.debug(msg=msg.format(frame.name, name_of_new_parent_link))
```